### DISCUSSION TITLE
Why isn't there a method to get the current route'pathname

### DISCUSSION THREAD

**Discussion by guaijie** [remix-run/react-router#11219] [Q&A]
Why isn't there a method to get the current route'pathname

**Comment by brophdawg11** [comment]
Do you mean [`useLocation().pathname`](https://reactrouter.com/en/main/hooks/use-location#locationpathname)?  

Or if you mean the `route.path` - you can get that via `useMatches()` if you are using `RouterProvider`, or you can do it via `matchRoutes(routes, useLocation().pathname)` in `<BrowserRouter>`

**Reply by guaijie** [reply]
I want to be able to get route's pathname in a non-React component
